File: server/php/src/Application/DTOs/GetPaymentDetails/ResponseDto.php
<?php

namespace OnlinePayments\ExampleApp\Application\DTOs\GetPaymentDetails;

use OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails\OperationOutput;
use OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails\HostedCheckoutSpecificOutput;
use OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails\PaymentOutput;
use OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails\PaymentStatusOutput;

final class ResponseDto
{
    /** @var OperationOutput[]|null Array of OperationOutput objects */
    public ?array $operations = null;

    public ?HostedCheckoutSpecificOutput $hostedCheckoutSpecificOutput = null;

    public ?PaymentOutput $paymentOutput = null;

    public ?string $status = null;

    public ?PaymentStatusOutput $statusOutput = null;

    public ?string $id = null;
}

File: server/ruby/app/business/dtos/get_payment_details/response_dto.rb
module Business
  module Dtos
    module GetPaymentDetails
      class ResponseDto
        attr_accessor :operations
        attr_accessor :hosted_checkout_specific_output
        attr_accessor :payment_output
        attr_accessor :status
        attr_accessor :status_output
        attr_accessor :id
        def initialize(
          operations: nil,
          hosted_checkout_specific_output: nil,
          payment_output: nil,
          status: nil,
          status_output: nil,
          id: nil
        )
          @operations = operations
          @hosted_checkout_specific_output = hosted_checkout_specific_output
          @payment_output = payment_output
          @status = status
          @status_output = status_output
          @id = id
        end
      end
    end
  end
end